File: 424.longest-repeating-character-replacement.py
#
# @lc app=leetcode id=424 lang=python3
#
# [424] Longest Repeating Character Replacement
#

# @lc code=start
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        if k == len(s):
            return len(s)
        tracked_k = k
        start_pos = 0
        end_pos = 0
        running_len = 1
        maxlen = 0
        
        
# @lc code=end

File: 424.longest-repeating-character-replacement.cpp
/*
 * @lc app=leetcode id=424 lang=cpp
 *
 * [424] Longest Repeating Character Replacement
 */
#include <cstdio>
#include <string>
#include <algorithm>
using namespace std;

// @lc code=start
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.size();
        int l = 0;
        int r = 0;
        int ans = 0;
        int hash[26]={0};
        int maxFre = 0;
        while(r<n) {
            hash[s[r]-'A']++;
            if(hash[s[r]-'A']>maxFre){
                maxFre = hash[s[r]-'A'];
            }
            while((r-l+1)-maxFre>k) {
                hash[s[l]-'A']--;
                l++;
            }
            ans = max(ans, r-l+1);
            r++;      
        }
        return ans;       
    }
};
// @lc code=end

int main() {
    return 0;
}